pkgs/tools/text/languagetool/default.nix

```diff
@@ -1,12 +1,29 @@
-{ lib, stdenv, fetchzip, jre, makeWrapper, nixosTests }:
-
+{ lib
+, stdenv
+, fetchzip
+, jre
+, makeWrapper
+, nixosTests
+, writeTextFile
+, extraSpelling ? { } # Custom spelling { en = ["multithreading" "quadtree"]; global=["Nixos"];}
+}:
+let
+  mkSpellFile = lang: data: writeTextFile {
+    name = "spelling_custom_" + lang + ".txt";
+    text = builtins.concatStringsSep "\n" data;
+  };
+  mkCustomSpellPath = lang: "$out/share/org/languagetool/resource/" +
```

Comment:
We can't load this from ``/etc/share/org/languagetool/resource`` and configure this via a nixos module?

Reply:
I do not understand your question 100%. But the spelling files need to be in the right directory or they will not get loaded, and there is also no config option like `--custom-spelling` or env variable to change the paths. 

What we could do is expose this option trough a nixos module as well. This is a (precompiled) Java package so it does not take long if people want to customize this.

Reply:
We can always patch the directory where the spelling files are searched for. Inside of the nix store is usually not the right location.



> This is a (precompiled) Java package

Maybe it should be build from source instead or we use dangling symlinks?

Reply:
If a patch were to be used to do this, this would be the place:

https://github.com/languagetool-org/languagetool/blob/466b24579d4e1dbac15148523c1b8841265a4629/languagetool-core/src/main/java/org/languagetool/rules/spelling/SpellingCheckRule.java#L413

But I do not have time for this now, so maybe only the version bump should be done for now. And if I have time in a couple of months, I could look into this.